Sources/Datadog/RUM/RUMMonitor/Scopes/RUMSessionScope.swift

```diff
@@ -89,6 +94,10 @@ internal class RUMSessionScope: RUMScope, RUMContextProvider {
         }
         lastInteractionTime = command.time
 
+        if shouldBeSampledOut {
+            return true
+        }
```

Comment:
if we already know that the session is discarded (sampled-out) at init method, why do we wait until the next command?
i thought `ApplicationScope` would be responsible for sampling:
```swift
// AppScope
process(cmd) {
  case is StartSession and shouldSample(at: rate):
    sessionScope = SessionScope(...)
  // if shouldSample(at: rate) returns false, don't create sessionScope
}
```
am i missing something?

Reply:
> if we already know that the session is discarded (sampled-out) at init method, why do we wait until the next command?

To let the session expire or time-out when any command arrives at certain time.

> ```swift
> // AppScope
> process(cmd) {
>   case is StartSession and shouldSample(at: rate):
>     sessionScope = SessionScope(...)
>   // if shouldSample(at: rate) returns false, don't create sessionScope
> }
> ```

This wouldn't be correct - if the Session scope is not created, how would we know that the RUM Session has expired or timed-out? Without this information, we wouldn't know that the RUM Session should re-created.

> i thought ApplicationScope would be responsible for sampling

This would mean that the `RUMApplicationScope` manages session creation, session sampling and session time-out/expiration. With current model, the responsibilities are separated:
* `RUMApplicationScope` - manages session creation,
* `RUMSessionScope` - manages its time-out/expiration and sampling.